pkg/machines/components/vmSnapshotsTab.jsx

```diff
@@ -0,0 +1,174 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from "moment";
+
+import cockpit from 'cockpit';
+import { vmId } from "../helpers.js";
+import { ListingTable } from "cockpit-components-table.jsx";
+import { Tooltip } from '@patternfly/react-core';
+import { InfoAltIcon } from '@patternfly/react-icons';
+
+import './vmSnapshotsTab.css';
+
+const _ = cockpit.gettext;
+
+function prettyTime(unixTime) {
+    moment.locale(cockpit.language, {
+        longDateFormat : {
+            LT: "HH:mm:ss",
+            L: "DD/MM/YYYY",
+        }
+    });
+    const yesterday = _("Yesterday");
+    const today = _("Today");
+    moment.locale(cockpit.language, {
+        calendar : {
+            lastDay : `[${yesterday}] LT`,
+            sameDay : `[${today}] LT`,
+            sameElse : "L LT"
+        }
+    });
+
+    return moment(Number(unixTime) * 1000).calendar();
+}
+
+class VmSnapshotsTab extends React.Component {
+    render() {
+        const { vm } = this.props;
+        const id = vmId(vm.name);
+
+        const emptyCaption = (<div className="no-snapshots">
+            <h3>{_("No snapshots")}</h3>
+            {_("Snapshots allow you to revert to an earlier state if something goes wrong")}
```

Comment:
@skobyda was this suggested by the designers?
I think empty state string is good enough, I would drop the second line.

Reply:
Yes. The design was discussed here and it was part of mockup for empty state. https://github.com/cockpit-project/cockpit/issues/13010

Reply:
@andreasn comment about that line: "But only if you take them!
If this string is aimed towards people unfamiliar with snapshot functionality, I the the text need to be a little bit more specific. Can't think of a good exact string right now, though."
What about "Previously taken snapshots allow you to revert to an earlier state if something goes wrong"

Reply:
Yeah, I like that.